azurerm/resource_arm_api_management_logger_test.go

```diff
@@ -0,0 +1,409 @@
+					resource.TestCheckResourceAttr(resourceName, "eventhub.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "eventhub.0.name"),
+					resource.TestCheckResourceAttrSet(resourceName, "eventhub.0.connection_string"),
+				),
+			},
+			{
+				Config: testAccAzureRMApiManagementLogger_basicApplicationInsights(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApiManagementLoggerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "buffered", "true"),
+					resource.TestCheckResourceAttr(resourceName, "eventhub.#", "0"),
+				),
+			},
+			{
+				Config: testAccAzureRMApiManagementLogger_basicEventHub(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApiManagementLoggerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "buffered", "true"),
+					resource.TestCheckResourceAttr(resourceName, "eventhub.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "eventhub.0.name"),
+					resource.TestCheckResourceAttrSet(resourceName, "eventhub.0.connection_string"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMApiManagementLogger_complete(t *testing.T) {
+	resourceName := "azurerm_api_management_logger.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApiManagementLoggerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApiManagementLogger_complete(ri, location, "Logger from Terraform test", "false"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApiManagementLoggerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "description", "Logger from Terraform test"),
+					resource.TestCheckResourceAttr(resourceName, "buffered", "false"),
+					resource.TestCheckResourceAttr(resourceName, "eventhub.#", "0"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{"application_insights.0.instrumentation_key"},
+			},
+		},
+	})
+}
+
+func TestAccAzureRMApiManagementLogger_completeUpdate(t *testing.T) {
+	resourceName := "azurerm_api_management_logger.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApiManagementLoggerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApiManagementLogger_complete(ri, location, "Logger from Terraform test", "false"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApiManagementLoggerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "description", "Logger from Terraform test"),
```

Comment:
since it's very similar, I think we could probably add a final step to the other `_update` test to add a description and then remove this test case?